ChainSafe/lodestar
# feat: add ephemery network

**Description**

Add ephemery as a network option, requiring additions to `packages/cli`, `packages/config` and `packages/prover`. 

This PR addresses the first, second and third tasks of [issue#6050](https://github.com/ChainSafe/lodestar/issues/6050), relating to:
- Add ephemery flag
- Use backup download URL (ephemery.dev)
- Use flag to override with custom path to a `genesis.ssz` file provided by user

**Steps to test or reproduce**

Checkout the branch:
```sh
git checkout atkinsonholly/ephemery-config
```

To run the ephemery network using the `genesis.ssz` download available at https://ephemery.dev/latest/genesis.ssz, run:
```sh
lodestar beacon --ephemery
```
which should show 
```sh
warn: Initializing from a stale checkpoint state vulnerable to long range attacks
warn: Checkpoint sync recommended
```

To run the ephemery network using the checkpoint sync flag (recommended), run:
```sh
lodestar beacon --network ephemery --checkpointSyncUrl https://checkpointz.bordel.wtf/
```
which should result in 
```sh
info: Initializing beacon from a valid checkpoint state
```

To run the ephemery network with a local genesis state file, run:
```sh
lodestar beacon ephemery --genesisStateFile path/to/genesis.ssz
```
